### ISSUE TITLE
Permission denied when trying to deploy

### ISSUE THREAD

**Issue by aymard229** [lorisleiva/laravel-deployer#70]
Hi everyone,
Am having that issues when trying to deploy my laravel app on my hosting server :
**mkdir: cannot create directory `.dep': Permission denied**
### Output
• done on [mydomain.com]
[localhost] > git rev-parse --abbrev-ref HEAD
[localhost] < master
✈︎ Deploying master on mydomain.com with basicm
 strategy
• done on [mydomain.com]
• done on [mydomain.com]
➤ Executing task deploy:prepare
[mydomain.com] > echo $0
[mydomain.com] < bash
[mydomain.com] > if [ ! -d / ]; then mkdir -p /; fi
[mydomain.com] > if [ ! -L //current ] && [ -d //current ]; then e
cho 'true'; fi
[mydomain.com] > cd / && if [ ! -d .dep ]; then mkdir .dep; fi
[mydomain.com] < mkdir: cannot create directory `.dep': Permissio
n denied
➤ Executing task deploy:failed
• done on [mydomain.com]
✔ Ok [0ms]
➤ Executing task deploy:unlock
[mydomain.com] > rm -f //.dep/deploy.lock
• done on [mydomain.com]
✔ Ok [6s 512ms]

### What I tried ?

- Changing my root folder chmod to 777 an deploy again but still have the same issue

Thanks for your time and your help

**Comment by aymard229** [comment]
I agree with you about setting permissions to 777. My hosting provider is Namecheap. My web root is www and have a symlink to public_html.
Both of them have this permissions :
**drwxr-x---  7 root_user nobody      4096 Nov 21 12:02 public_html
lrwxrwxrwx  1 root_user root_user  11 May 22  2018 www -> public_html**

**Comment by stelgenhof** [comment]
Thanks! Quick question: If your webroot is `www` (can I assume it is `/var/www`?) why do you have this symlink to `public_html`?

The issue is actually that the user account you are using with Deployer (SSH account) has not the correct permissions to write in the target directory. There are various options to overcome that depending on your situation. I used the following steps as I have multiple sites on my server:

https://www.digitalocean.com/community/tutorials/how-to-host-multiple-websites-securely-with-nginx-and-php-fpm-on-ubuntu-14-04

(The tutorial is for Ubuntu 14.04 but should work for other versions as well). 

Cheers! Sacha

**Comment by aymard229** [comment]
I can't tell you exactly if it's `/var/www`. The symlink was already done by Namecheap when I hve purchased my hosting plan. Am not on VPS but shared hosting, anyway, am gonna follow the tutorial you provide to me and update this.

Thanks !

**Comment by stelgenhof** [comment]
Let me know how things are going. Often with shared hosting, providers give you a starting point for your setup. In most cases it can be changed/removed as well (if you feel like it).
